### PR TITLE
Redis! Don't quit!

### PR DESCRIPTION
**Pull request by digitarald**
Next up:

- [ ] Travis! Don't fail to run tests!
- [ ] Heroku! Don't fail to build!

### CODE REVIEW

**Comment by digitarald** [comment]
Tests are stuck with last line `[23:56:14] Finished 'build' after 141 ms`, but no other indication of failure in the logs.

**Comment by digitarald** [comment]
Can reproduce locally, tests don't start.

**Comment by digitarald** [comment]
We probably have a redis connection open in the root of the gulpfile: https://github.com/sindresorhus/gulp-imagemin/issues/20#issuecomment-48280349

**Comment by zalun** [comment]
If it hangs it means the connection to redis isn't closed.

File: tests/apiUnit.js [hunk 1]
```diff
@@ -37,18 +37,14 @@ define((require) => {
   function flushDB(client) {
     return redis.flushdb(client);
   }
-  function flushQuitDB(client) {
-    return flushDB(client).then(() => digger.quitClient(5));
-  }
-
 
   bdd.describe('digger module', () => {
```
Comment (zalun) [inline tests/apiUnit.js]: There is a need to quit after all tests are done
```javascript
bdd.after(() => digger.quitClient(5));
```

**Comment by zalun** [comment]
I see you don't quit database at all. That would mean gulp process will never end. It waits for redis connection to be finished

**Comment by zalun** [comment]
building status quite often creates new connection.
After adding `console.log`s at the beginning of `readJSON` and in `getRequest` just before creating new redis connection:
```
[15:23:13] Starting 'build:status'...
XXX: readJSON
XXX: creating new connection
XXX: readJSON
XXX: creating new connection
XXX: readJSON
XXX: creating new connection
XXX: readJSON
XXX: creating new connection
XXX: readJSON
XXX: creating new connection
[15:23:13] Starting 'build:css'...
[15:23:13] Starting 'build:js'...
[15:23:14] Finished 'build:engines' after 296 ms
[15:23:14] Finished 'build:tabzilla' after 292 ms
[15:23:14] Finished 'build:routes' after 305 ms
[15:23:14] Starting 'build:app'...
[15:23:14] Finished 'build:app' after 95 ms
[15:23:14] Finished 'build:css' after 724 ms
[15:23:14] Finished 'build:js' after 863 ms
[15:23:14] Finished 'build:root' after 910 ms
XXX: readJSON
[... a lot of XXX: readJSON ...]
XXX: readJSON
```

**Comment by zalun** [comment]
I've created #423 to show what I think is needed...
Unfortunately it involves opening and closing the redis connection at will

**Comment by digitarald** [comment]
test and build are fixed already

**Pull request closed without merging**